migrator/migrations/m_198_to_m_199_policy_description_and_criteria_updates/migration_impl.go

```diff
@@ -0,0 +1,99 @@
+package m198tom199
+
+import (
+	"context"
+	"embed"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/migrator/migrations/m_198_to_m_199_policy_description_and_criteria_updates/schema"
+	"github.com/stackrox/rox/migrator/migrations/policymigrationhelper"
+	"github.com/stackrox/rox/migrator/types"
+	"gorm.io/gorm"
+)
+
+var (
+
+	//go:embed policies_before_and_after
+	policyDiffFS embed.FS
+
+	// Update only if the existing name, description and policy sections haven't changed.
+	fieldsToCompare = []policymigrationhelper.FieldComparator{
+		policymigrationhelper.NameComparator,
+		policymigrationhelper.DescriptionComparator,
+		policymigrationhelper.PolicySectionComparator,
+	}
+
+	policyDiffs = []policymigrationhelper.PolicyDiff{
+		{
+			FieldsToCompare: fieldsToCompare,
+			PolicyFileName:  "exec-adduser.json",
+		},
+		{
+			FieldsToCompare: fieldsToCompare,
+			PolicyFileName:  "exec-systemctl.json",
+		},
+		{
+			FieldsToCompare: fieldsToCompare,
+			PolicyFileName:  "no_resources_specified.json",
+		},
+		{
+			FieldsToCompare: fieldsToCompare,
+			PolicyFileName:  "root_user.json",
+		},
+		{
+			FieldsToCompare: fieldsToCompare,
+			PolicyFileName:  "sensitive_files.json",
+		},
+
+		{
+			FieldsToCompare: fieldsToCompare,
+			PolicyFileName:  "automount_service_account_token.json",
+		},
+		{
+			FieldsToCompare: fieldsToCompare,
+			PolicyFileName:  "host_pids.json",
+		},
+		{
+			FieldsToCompare: fieldsToCompare,
+			PolicyFileName:  "host_network.json",
+		},
+	}
+)
+
+func migrate(database *types.Databases) error {
+
+	db := database.GormDB
+	return policymigrationhelper.MigratePoliciesWithDiffsAndStoreV2(
+		policyDiffFS,
+		policyDiffs,
+		// Get policy with specified id
+		func(ctx context.Context, id string) (*storage.Policy, bool, error) {
+			var foundPolicy schema.Policies
+			result := db.WithContext(ctx).Table(schema.PoliciesTableName).Where(&schema.Policies{ID: id}).First(&foundPolicy)
```

Comment:
can we add this as a generic function in migration helper?

Reply:
Unfortunately no. The `schema` package there is specific to each migration. It's the frozen schema at the point of the migration. By moving it to the helper, we wouldn't be able to freeze the schema every migration.